administrator/components/com_installer/src/Model/UpdateModel.php

```diff
@@ -453,7 +453,23 @@ private function install($update)
         $package = InstallerHelper::unpack($tmp_dest . '/' . $p_file);
 
         if (empty($package)) {
-            $app->enqueueMessage(Text::sprintf('COM_INSTALLER_UNPACK_ERROR', $p_file), 'error');
+            $dispatcher = Factory::getApplication()->getDispatcher();
```

Comment:
Please use $this->getDispatcher()
```suggestion
            $dispatcher = $this->getDispatcher();
```